Reference Material/harcourt_db/ipbans.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/python
+import pyshark
+import subprocess
+import sqlite3
+from sqlite3 import Error
+
+def create_connection(db_file):
+    """ create a database connection to the SQLite database
+        specified by the db_file
+    :param db_file: database file
+    :return: Connection object or None
+    """
+    try:
+        conn = sqlite3.connect(db_file)
+        return conn
+    except Error as e:
+        print(e)
+ 
+    return None
+ 
+def read_bans(conn):
+    """
+    Query all rows in the tasks table
+    :param conn: the Connection object
+    :return:
+    """
+    cur = conn.cursor()
+    cur.execute("SELECT * FROM ipblacklist")
+ 
+    rows = cur.fetchall()
+ 
+    for row in rows:
+        lst2.append(row)
+	
+capture = pyshark.LiveCapture(interface='eth0')
+capture.sniff(timeout=50)
+#print len(capture)
+lst1=[]
+#IP bans
+lst2=[]
+
+def blockip(ip):
+	cmd="/sbin/iptables -A INPUT -s "+ip+" -j DROP"
+	print cmd
+	subprocess.call(cmd,shell=True)
+ 
+def blockIP1(ip):
+	cmd="/sbin/iptables -A INPUT -s "+ip+" -j DROP"
+	subprocess.call(cmd, shell=True)
+	subprocess.call("kill -9 $(/usr/bin/pgrep dumpcap)", shell=True)
```

Comment:
Use [`os.kill`](https://docs.python.org/3/library/os.html#os.kill) instead. Also, SIGKILLing these things is just a horrible approach. This blows up if somebody is using the same executable for something else, and it's 100% an anti pattern and a problem elsewhere that you have to do this at all.